tests/integ/modin/series/test_reindex.py

```diff
@@ -328,15 +325,26 @@ def perform_reindex(series):
     )
 
 
-@sql_count_checker(query_count=0)
-def test_reindex_index_non_overlapping_different_types_index_negative():
+@sql_count_checker(query_count=1, join_count=2)
+def test_reindex_index_non_overlapping_different_types_index():
+    date_index = native_pd.date_range("1/1/2010", periods=6, freq="D")
+    native_series = native_pd.Series(
+        {"prices": [100, 101, np.nan, 100, 89, 88]}, index=date_index
+    )
     date_index = pd.date_range("1/1/2010", periods=6, freq="D")
     snow_series = pd.Series(
         {"prices": [100, 101, np.nan, 100, 89, 88]}, index=date_index
     )
 
-    with pytest.raises(SnowparkSQLException, match=".*Timestamp 'A' is not recognized"):
-        snow_series.reindex(list("ABC")).to_pandas()
+    def perform_reindex(series):
```

Comment:
We could probably just do this as a lambda right? Both branches of the if statement lead to the same thing?